provider/provider.go

```diff
@@ -10,6 +10,37 @@ import (
 	"github.com/go-spatial/tegola/internal/log"
 )
 
+type providerType uint8
+
+const (
```

Comment:
I think only the `providerType` constants should be defined, and be bit flags. And the filters are implemented with a function:
```go
func ProviderFilterInclude(filters ...providerType) providerFilter {
    ret := uint8(0)
    for _, v := range filters {
        ret |= uint8(v)
    }
    return providerFilter(ret)
}
```

Having two sets of constants that essentially refer to the same information could be confusing